### PR TITLE
Nodes

### PR DESCRIPTION
**Pull request by eggrobin**
apsides.hpp etc. should have a proper name, but I can't find anything that covers apsides and nodes...

### CODE REVIEW

File: physics/apsides_body.hpp [hunk 1]
```diff
@@ -105,6 +109,62 @@ void ComputeApsides(Trajectory<Frame> const& reference,
...
 
+template<typename Frame>
+void ComputeNodes(typename DiscreteTrajectory<Frame>::Iterator begin,
+                  typename DiscreteTrajectory<Frame>::Iterator end,
+                  Vector<double, Frame> const& north,
+                  DiscreteTrajectory<Frame>& ascending,
+                  DiscreteTrajectory<Frame>& descending) {
+  std::experimental::optional<Instant> previous_time;
+  std::experimental::optional<DegreesOfFreedom<Frame>>
+      previous_degrees_of_freedom;
```
Comment (pleroy) [inline physics/apsides_body.hpp]: Mostly harmless.

File: physics/apsides_test.cpp [hunk 2]
```diff
@@ -136,6 +140,124 @@ TEST_F(ApsidesTest, ComputeApsidesDiscreteTrajectory) {
...
+      t0 + 10 * JulianYear,
+      Ephemeris<World>::AdaptiveStepParameters(
+          DormandElMikkawyPrince1986RKN434FM<Position<World>>(),
+          std::numeric_limits<std::int64_t>::max(),
+          1e-3 * Metre,
+          1e-3 * Metre / Second),
+      Ephemeris<World>::unlimited_max_ephemeris_steps,
+      /*last_point_only=*/false);
+
+  Vector<double, World> const north({0,0,1});
```
Comment (pleroy) [inline physics/apsides_test.cpp]: Space after commas.

File: physics/apsides_body.hpp (hunk 1 above)
Comment (eggrobin) [inline physics/apsides_body.hpp]: Done.

File: physics/apsides_body.hpp [hunk 3]
```diff
@@ -105,6 +109,62 @@ void ComputeApsides(Trajectory<Frame> const& reference,
...
+  std::experimental::optional<Length> previous_z;
+  std::experimental::optional<Speed> previous_z_speed;
+
+  for (auto it = begin; it != end; ++it) {
+    Instant const time = it.time();
+    DegreesOfFreedom<Frame> const degrees_of_freedom = it.degrees_of_freedom();
```
Comment (eggrobin) [inline physics/apsides_body.hpp]: Done.

File: physics/apsides_body.hpp [hunk 4]
```diff
@@ -105,6 +109,62 @@ void ComputeApsides(Trajectory<Frame> const& reference,
...
+    Length const z =
+        (degrees_of_freedom.position() - Frame::origin).coordinates().z;
+    Speed const z_speed = degrees_of_freedom.velocity().coordinates().z;
+
+    if (previous_z && Sign(z) != Sign(*previous_z)) {
+      CHECK(previous_time && previous_z && previous_z_speed);
```
Comment (eggrobin) [inline physics/apsides_body.hpp]: Done.

**Comment by pleroy** [comment]
retest this please

**Pull request merged**